cdap-validation/bin/test_helpers.py

```diff
@@ -0,0 +1,71 @@
+import os
+import base64
+import urllib2
+import json
+import sys
+
+def onetime_auth(host,info):
+    # this should work with all, but when authentication is not set up correctly, this fails (http 403)
+    passman = urllib2.HTTPPasswordMgrWithDefaultRealm() # this creates a password manager
+    passman.add_password(None, host, info['username'], info['password'])
+    authhandler = urllib2.HTTPBasicAuthHandler(passman) # create the AuthHandler
+    opener = urllib2.build_opener(authhandler)
+    urllib2.install_opener(opener)
+    return urllib2
+
+def run_request(host,info):
+    user = info['username']
+    passwd = info['password']
+    try:
+        handle = urllib2.urlopen(host)
+        return handle
+    except: #except HTTPError, h: ### would like to do something like this
+        # try a different way (encoding headers)
+        req = urllib2.Request(host)
+        base64string = base64.encodestring('%s:%s' % (user, passwd))[:-1]
+        authheader =  "Basic %s" % base64string
+        req.add_header("Authorization", authheader)
+        try:
+            handle = urllib2.urlopen(req)
+            return handle
+        except IOError, e:
+            # here we shouldn't fail if the username/password is right
+            print "It looks like the username or password is wrong."
+            return 'noapi'
+
+def write_config_file(config, dir, file):
```

Comment:
this helper method just writes any arbitrary string to a file, and so does not need ``config`` in its name.